src/Compilers/CSharp/Portable/Symbols/Tuples/TupleTypeSymbol.cs

```diff
@@ -94,13 +94,7 @@ internal sealed class TupleTypeSymbol : WrappedNamedTypeSymbol
         {
             Debug.Assert(!shouldCheckConstraints || (object)syntax != null);
             Debug.Assert(elementNames.IsDefault || elementTypes.Length == elementNames.Length);
-
-            int numElements = elementTypes.Length;
-
-            if (numElements <= 1)
-            {
-                throw ExceptionUtilities.Unreachable;
-            }
+            Debug.Assert(elementTypes.Length > 1 || elementNames.IsDefault);
```

Comment:
>Debug.Assert(elementTypes.Length > 1 || elementNames.IsDefault); [](start = 12, length = 64)

This change feels unexpected, there is no such thing as one element tuples in source. Either we are supporting scenarios that shouldn't be supported, or we are misusing this helper.

Reply:
Yes, as we discussed I will disallow comparison of two `ValueTuple'1` instances, as they are not tuples. Then I'll revert this here.

---
In reply to: [170487525](https://github.com/dotnet/roslyn/pull/23356#discussion_r170487525) [](ancestors = 170487525)